megazords/ios-rust/README.md

```diff
@@ -0,0 +1,108 @@
+# XCFramework build for distributing Rust code on iOS
+
+This directory contains the logic for compiling all of our Rust code into a binary
+artifact that can be easily distributed to iOS consumers. If you run the following
+script:
+
+```
+$> ./build-xcframework.sh
+```
+
+Then it should produce a file named `MozillaRustComponents.xcframework.zip` that
+contains:
+
+* The compiled Rust code for all the crates listed in `Cargo.toml`, as a static library,
+* along with their corresponding C header files and Swift module maps,
+* built for all our target iOS platforms, as an "XCFramework" bundle.
+
+The resulting `.zip` is suitable for consumption as a Swift Package binary dependency.
+
+## What's here?
+
+In this directory we have:
+
+* A Rust crate that serves as the "megazord" for our iOS distributions; it basically depends
+  on all the Rust Component crates and re-exports their public APIs.
+* Some skeleton files for building an XCFramework:
+    * `module.modulemap` is a "module map", which tells the Swift compiler how to use C-level APIs.
+    * `MozillaRustComponents.h` is an "umbrella header", used by the module map as a shortcut
+      to specify all the available header files.
+    * `Info.plist` specified metadata about the resulting XCFramework, such as the available
+      architectures and their subdirectories.
+* The `build-xcframework.sh` script which knows how to stitch things together into a full
+  XCFramework bundle.
+    * The XCFramework format is not well documented; briefly:
+        * It's a directory containing resources compiled for multiple target architectures,
+          typically distributed as `.zip` file.
+        * The top-level directory contains a subdirectory per architecture, and an `Info.plist`
+          file that says what things live in which directory.
+        * Each subdirectory contains a `.framework` directory for that architecture. There
+          are notes on the layout of an individual `.framework` in the links below.
+
+It's a little unusual that we're building the XCFramework by hand, rather than defining it
+as the build output of an Xcode project. It turns out to be simpler for our purposes, but
+does risk diverging from the expected format if Apple changes the detailts of XCFrameworks
+in future Xcode releases.
+
+## Adding crates
+
+To add a new crate to the distribution:
+
+1. Update its `uniffi.toml`, if any, to include the following settings:
+    ```
+    [bindings.swift]
+    ffi_module_name = "MozillaRustComponents"
+    ffi_module_filename = "<crate_name>FFI"
+    ```
+1. Add it as a dependency in `Cargo.toml`.
+1. Add a `pub use` declaration for it in `./src/lib.rs`.
+1. Add logic to `build-xcframework.sh` to copy or generate its header file into the build.
+1. Add a `#import` for its header file to `MozillaRustComponents.h`
```

Comment:
Do we want to be explicit that the `#import` should follow the `#if canImport(...)` pattern? I think it's easy enough to figure out, especially if someone looks at one of the already existing components, but just thinking out loud

Reply:
Never mind! I have this the other way around 🤦 🤦